internal/core/services/room_manager/room_manager.go

```diff
@@ -98,10 +98,24 @@ func (m *RoomManager) DeleteRoom(ctx context.Context, gameRoom *game_room.GameRo
 		return fmt.Errorf("failed to delete instance on the runtime: %w", err)
 	}
 
-	err = m.SetRoomStatus(ctx, gameRoom, game_room.GameStatusTerminating)
+	terminatingGameRoom := *gameRoom
+	terminatingGameRoom.Status = game_room.GameStatusTerminating
+	err = m.setRoomStatus(ctx, gameRoom, &terminatingGameRoom)
 	if err != nil {
 		return fmt.Errorf("failed to update game room status to terminating: %w", err)
 	}
 
 	return nil
 }
+
+func (m *RoomManager) UpdateRoom(ctx context.Context, gameRoom *game_room.GameRoom) error {
+	currentGameRoom, err := m.roomStorage.GetRoom(ctx, gameRoom.SchedulerID, gameRoom.ID)
+	if err != nil {
+		return fmt.Errorf("unable to fetch game room from storage: %w", err)
+	}
+	err = m.setRoomStatus(ctx, currentGameRoom, gameRoom)
```

Comment:
setRoomStatus is doing way more than just setting the room status. If your intent here is just to apply the game room state machine, I would rename it to `validateStatusTransition` and let the update itself living here.

Reply:
@luizmiranda7  this isolation definitely makes more sense than the actual implementation! going to change it

Reply:
solved in a30c31b